addons/hr_skills/models/hr_skill_type.py

```diff
@@ -19,3 +17,12 @@ def _get_default_color(self):
     skill_ids = fields.One2many('hr.skill', 'skill_type_id', string="Skills")
     skill_level_ids = fields.One2many('hr.skill.level', 'skill_type_id', string="Levels")
     color = fields.Integer('Color', default=_get_default_color)
+
+    def copy(self, default=None):
+        new_types = super().copy(default)
+        for old_type, new_type in zip(self, new_types):
+            new_skills_level = old_type.skill_level_ids.copy({"skill_type_id": new_type.id})
+            new_type.write({
+                'skill_level_ids': [(Command.set(new_skills_level.ids))]
+                })
+        return new_types
```

Comment:
Could be achievable by setting ``copy=True`` on the field ``skill_level_ids`` nope ?